### PR TITLE
Add `tls-verify` configuration to container-image-podman

### LINKED ISSUE #43486: Support to skip TLS verification when image is pushed for private HTTPS registries
## Feature request

The command pushing the image build to a private registry (e.g. localhost:5000, gitea.cnoe.localtest.me:8443, etc) should allow to skip to verify the TLS certificate to avoid to get the following error 

```
[INFO] --- quarkus:3.14.4:image-push (default-cli) @ code-with-quarkus ---
[INFO] [io.quarkus.kubernetes.deployment.PropertyUtil] Kubernetes manifests are generated with 'The container port http' having default value '8080'. The app and manifests will get out of sync if the property 'quarkus.http.port' is changed at runtime.
[INFO] Testcontainers version: 1.20.1
[INFO] Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
[INFO] Found Docker environment with Environment variables, system properties and defaults. Resolved dockerHost=unix:///var/run/docker.sock
[INFO] Docker host IP address is localhost
[INFO] Connected to docker: 
  Server Version: 5.0.3
  API Version: 1.41
  Operating System: fedora
  Total Memory: 5487 MB
[INFO] Image pull policy will be performed by: DefaultPullPolicy()
[INFO] Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
[INFO] Checking the system...
[INFO] ✔︎ Docker server version should be at least 1.6.0
[INFO] [io.quarkus.deployment.util.ExecUtil] Error response from daemon: {"errorDetail":{"message":"trying to reuse blob sha256:dd5e77a90e609b328f2e49aa60e50bd8837e505c157060c337725413ccf449f1 at destination: pinging container registry gitea.cnoe.localtest.me:8443: Get \"https://gitea.cnoe.localtest.me:8443/v2/\": tls: failed to verify certificate: x509: certificate signed by unknown authority"},"error":"trying to reuse blob sha256:dd5e77a90e609b328f2e49aa60e50bd8837e505c157060c337725413ccf449f1 at destination: pinging container registry gitea.cnoe.localtest.me:8443: Get \"https://gitea.cnoe.localtest.me:8443/v2/\": tls: failed to verify certificate: x509: certificate signed by unknown authority"}
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.827 s
[INFO] Finished at: 2024-09-25T08:45:58+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.14.4:image-push (default-cli) on project code-with-quarkus: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.container.image.docker.deployment.DockerProcessor#dockerBuildFromJar threw an exception: java.lang.RuntimeException: Execution of 'docker push gitea.cnoe.localtest.me:8443/giteaadmin/code-with-quarkus:1.0.0-SNAPSHOT' failed. See docker output for more details
[ERROR]         at io.quarkus.container.image.docker.common.deployment.CommonProcessor.containerRuntimeException(CommonProcessor.java:254)
[ERROR]         at io.quarkus.container.image.docker.common.deployment.CommonProcessor.pushImage(CommonProcessor.java:225)
```

using the following commands
```
podman machine start
export DOCKER_HOST="unix:///var/run/docker.sock"            
./mvnw install -Dquarkus.container-image.build=true \
   -Dquarkus.container-image.push=true \
   -Dquarkus.container-image.insecure=true \
   -Dquarkus.container-image.image=gitea.cnoe.localtest.me:8443/giteaadmin/code-with-quarkus:1.0.0-SNAPSHOT

or

quarkus image push \
  --registry gitea.cnoe.localtest.me:8443 \
  --group giteaadmin \
  --name code-with-quarkus \
  --tag 1.0.0-SNAPSHOT
```

Of course, that works using:
```
podman build -f src/main/docker/Dockerfile.jvm -t gitea.cnoe.localtest.me:8443/giteaadmin/code-with-quarkus:1.0.0-SNAPSHOT .
podman push gitea.cnoe.localtest.me:8443/giteaadmin/code-with-quarkus:1.0.0-SNAPSHOT --tls-verify=0
```

**Remark**: We should also offer a new parameter to the CLI to allow to consume the image URL directky without having to pass registry, group, name and tag arguments
```
quarkus image push gitea.cnoe.localtest.me:8443/giteaadmin/code-with-quarkus:1.0.0-SNAPSHOT
```

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by geoand**
- Closes: #43486

### CODE REVIEW

File: extensions/container-image/container-image-podman/deployment/src/main/java/io/quarkus/container/image/podman/deployment/PodmanConfig.java [hunk 1]
```diff
@@ -16,4 +17,10 @@ public interface PodmanConfig extends CommonConfig {
      * https://docs.podman.io/en/latest/markdown/podman-build.1.html#platform-os-arch-variant
      */
     Optional<List<String>> platform();
+
+    /**
+     * Require HTTPS and verify certificates when contacting registries
+     */
+    @WithDefault("true")
+    boolean tlsVerify();
```
Comment (cmoulliard) [inline extensions/container-image/container-image-podman/deployment/src/main/java/io/quarkus/container/image/podman/deployment/PodmanConfig.java:25]: Will this parameter added automatically to the doc published here: https://quarkus.io/guides/container-image#PodmanOptions ?

File: extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java [hunk 2]
```diff
@@ -141,7 +141,7 @@ protected String createContainerImage(ContainerImageConfig containerImageConfig,
         if (!useBuildx && pushContainerImage) {
             // If not using buildx, push the images
             loginToRegistryIfNeeded(containerImageConfig, containerImageInfo, executableName);
-            pushImages(containerImageInfo, executableName);
+            pushImages(containerImageInfo, executableName, dockerConfig);
```
Comment (cmoulliard) [inline extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java:144]: Why do you use as word "dockerConfig" and not "containerConfig" ?

**Review by cmoulliard — COMMENTED** [review]
Is there an integration test validating such a change ?

File: extensions/container-image/container-image-podman/deployment/src/main/java/io/quarkus/container/image/podman/deployment/PodmanConfig.java (hunk 1 above)
Comment (geoand) [reply extensions/container-image/container-image-podman/deployment/src/main/java/io/quarkus/container/image/podman/deployment/PodmanConfig.java:25]: Yes

File: extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java (hunk 2 above)
Comment (geoand) [reply extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java:144]: That's what it's already called

**Comment by cmoulliard** [comment]
PR approved

**Review by iocanel — APPROVED** [review]
LGTM

**Pull request merged**